### DISCUSSION TITLE
Does Redux Persist allow for `Date` objects?

### DISCUSSION THREAD

**Discussion by trajano** [reduxjs/redux-toolkit#4014] [Q&A] [answered]
Just wondering does Redux Persist allow for `Date` objects or are they considered non-serializable?  In which case I should just store the `toISOString` value and convert it to a `Date` object using code?

**Comment by markerikson** [comment, marked as answer]
Redux Persist isn't part of RTK, but yeah, `Date` class instances would fall under the "not serializable" category, and you're better off keeping date values in the store as numeric timestamps or strings:

- https://redux.js.org/style-guide/#do-not-put-non-serializable-values-in-state-or-actions

**Reply by trajano** [reply]
A small follow up question...
what is considered "serializable" I am presuming: `string`, `number` and `boolean` are the only ones?  Or is `boolean` not even considered and it should just be number which we translate from `0/1` to `false/true` ?

**Reply by markerikson** [reply]
Basically just whatever will be returned natively from `JSON.parse()`.  So, plain objects/arrays, and primitive values.

**Reply by EskiMojo14** [reply]
anything you can do `JSON.parse(JSON.stringify(value))` and get an equivalent result from is serialiseable as far as redux-persist is concerned. primitives, arrays, and POJOs are all serialisable.
